git-guides/git-push.md

```diff
@@ -0,0 +1,57 @@
+## General Specifications
+
+# Git Push
+
+```sh
+git push
+```
+
+`git push` uploads all local branch commits to the corresponding remote branch.
+
+## What Does `git push` Do?
+
+`git push` updates the remote branch with local commits. It is one of the four commands in Git that prompts interaction with the remote repository. You can also think of `git push` as _update_ or _publish_. 
+
+By default, `git push` only updates the corresponding branch on the remote. So, if you are checked out to the `master` branch when you execute `git push`, then only the `master` branch will be updated. It's always a good idea to use `git status` to see what branch you are on before pushing to the remote.
+
+## How to Use `git push`
+
+After you make and commit changes locally, you can share them with the remote repository using `git push`. Pushing changes to the remote makes your commits accessible to others who you may be collaborating with. This will also update any open pull requests with the branch that you're working on. 
+
+As best practice, it's important to run the `git pull` command before you push any new changes to the remote branch. This will update your local branch with any new changes that may have been pushed to the remote from other contributors. Pulling before you push can reduce the amount of merge conflicts you create on GitHub - allowing you to resolve them locally before pushing your changes to the remote branch. 
+
+### Common usages and options for `git push`
+
+* `git push -f`: description
+* `git push -u origin [branch]`: description
+* `git push --all`: description
+* `git push --tags`: description
```

Comment:
```suggestion
* `git push -f`: description
* `git push -u origin [branch]`: description
* `git push --all`: description
* `git push --tags`: Publish tags that aren't yet in the remote repository
```

I'm not sure if `description` here is a placeholder. Do each of these need a description to be added?

Reply:
Good catch! 🙈 Yes, those are all placeholders.